helixbass/tsc-rust
# Parse + bind object literal

In this PR:
- parse + bind an object literal

To test:
If you run `cargo run tmp.tsx` against a source file that contains:
```
interface Foo {
  bar: number
}

const a: Foo = {
  baz: 1,
}
```
then in the `post-binding:` debug-logging you should see in the `SourceFile` `statements` first an `InterfaceDeclaration` (per previous PR) and then a `VariableStatement`/`VariableDeclaration` where the `initializer` is an `ObjectLiteralExpression` whose `properties` contains a single `PropertyAssignment` with `name: "Baz"` and `initializer` a `NumericLiteral` with `text: "1"` and whose `type_` is a `TypeReferenceNode` with `type_name: "Foo"`

Based on `parse-interface-declaration`
